docs: update static-deploy.md to support custom domains with GitHub Pages

mark-mxwl opened the pull request:
### Description

Added documentation to support custom domains with GitHub Pages. Currently, the section only addresses deployments to GitHub Pages repository URLs. Additional steps are required to correctly configure a build for GitHub Pages custom domains.

### Additional context

---

### What is the purpose of this pull request?

- [ ] Bug fix
- [ ] New Feature
- [X] Documentation update
- [ ] Other

### Before submitting the PR, please make sure you do the following

- [X] Read the [Contributing Guidelines](https://github.com/vitejs/vite/blob/main/CONTRIBUTING.md), especially the [Pull Request Guidelines](https://github.com/vitejs/vite/blob/main/CONTRIBUTING.md#pull-request-guidelines).
- [X] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [X] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [X] Update the corresponding documentation if needed.
- [ ] Ideally, include relevant tests that fail without this PR but pass with it.

mark-mxwl committed:
Update static-deploy.md

Added documentation to support custom domains with GitHub Pages. Currently, the section only addresses deployments to GitHub Pages repository URLs. Additional steps are required to correctly configure a build for GitHub Pages custom domains.

sapphi-red commented on docs/guide/static-deploy.md:
````diff
@@ -118,6 +118,14 @@ Now the `preview` command will launch the server at `http://localhost:8080`.
            uses: actions/deploy-pages@v2
    ```
 
+3. If you're deploying to GitHub Pages with a custom domain (eg. `www.example.com`), the following steps may need to be taken to avoid errors.
+
+   Set the `base` in `vite.config.js` to `'/'`.
````
I think we can change this part above instead.

> If you are deploying to `https://<USERNAME>.github.io/`, you can omit `base` as it defaults to `'/'`.

mark-mxwl replied:
Thanks for the suggestion! I agree. The initial statement can be revised to include custom domains with GitHub Pages, and to improve clarity. I'll make the adjustments.

sapphi-red commented on docs/guide/static-deploy.md:
```diff
@@ -118,6 +118,14 @@ Now the `preview` command will launch the server at `http://localhost:8080`.
+
+   Set the `base` in `vite.config.js` to `'/'`.
+
+   Ensure that your `index.html` correctly points to the root file by adding a `.` in front of the path (eg. `./src/main.jsx`).
+
+   Set the `homepage` property in your `package.json` to `https://<USERNAME>.github.io/<REPO>/`.
```
I don't think these are necessary. It should work without these changes.

mark-mxwl replied:
I ran an additional test to check this, and I agree that these changes are unnecessary if the base is correctly set. Will omit.

sapphi-red reviewed (dismissed):
Thanks!

mark-mxwl committed:
Merge branch 'vitejs:main' into patch-1

mark-mxwl committed:
Update static-deploy.md

Consolidated previous additions into an existing item, and revised language for clarity. Made a minor revision to the following item to keep example syntax consistent.

sapphi-red commented on docs/guide/static-deploy.md:
```diff
@@ -58,9 +58,9 @@ Now the `preview` command will launch the server at `http://localhost:8080`.
 
 1. Set the correct `base` in `vite.config.js`.
 
-   If you are deploying to `https://<USERNAME>.github.io/`, you can omit `base` as it defaults to `'/'`.
```
Do you have any reasons why you changed to suggest setting `base` explicitly?

sapphi-red replied:
I see. Including the option of removing the base would be nice.

mark-mxwl committed:
Update static-deploy.md

Added language to include the possibility of removing the base property in the given instance.

mark-mxwl commented:
Thanks @sapphi-red for all your help, and for being so responsive!
